sys/net/application_layer/dhcpv6/client.c

```diff
@@ -716,6 +848,100 @@ static bool _parse_reply(uint8_t *rep, size_t len)
+                         ia_na_len -= _opt_len(ia_na_opt),
+                         ia_na_opt = _opt_next(ia_na_opt)) {
+                        if (ia_na_len > ia_na_orig_len) {
+                            DEBUG("DHCPv6 client: IA_NA options overflow option "
+                                  "boundaries\n");
+                            return false;
+                        }
+                        switch (byteorder_ntohs(ia_na_opt->type)) {
+                            case DHCPV6_OPT_STATUS: {
+                                if (!_check_status_opt((dhcpv6_opt_status_t *)ia_na_opt)) {
+                                    continue;
+                                }
+                                break;
+                            }
+                            case DHCPV6_OPT_IAADDR: {
+                                dhcpv6_opt_iaaddr_t *this_iaaddr = (dhcpv6_opt_iaaddr_t *)ia_na_opt;
+                                if ((!lease->leased) ||
+                                    (iaaddr == NULL)) {
+                                    /* only take first address for now */
+                                    iaaddr = this_iaaddr;
+                                }
+                                break;
+                            }
+                            default:
+                                break;
+                        }
+                    }
+                    na_t1 = byteorder_ntohl(ia_na->t1);
+                    na_t2 = byteorder_ntohl(ia_na->t2);
+                    if ((na_t1 != 0) && (na_t2 != 0) &&
+                        ((server.t1 == 0) || (server.t1 >= na_t1)) &&
+                        ((t2 == 0) || (t2 >= na_t2))) {
+                        server.t1 = na_t1;
+                        t2 = na_t2;
+                        _schedule_t1_t2();
+                    }
+                    if ((iaaddr != NULL)) {
+                        uint32_t valid = byteorder_ntohl(iaaddr->valid);
+                        uint32_t pref = byteorder_ntohl(iaaddr->pref);
+                        ipv6_addr_t *addr = &iaaddr->addr;
+                        netif_t *netif = netif_get_by_id(lease->parent.ia_id.info.netif);
+                        char addr_str[IPV6_ADDR_MAX_STR_LEN];
+
+                        if (&lease->addr != NULL &&
+                            !ipv6_addr_equal(&lease->addr, addr)) {
+                            /* A different address has been leased to the client */
+                            netif_set_opt(netif, NETOPT_IPV6_ADDR_REMOVE, 64 << 8, &lease->addr, sizeof(lease->addr));
+                        }
+
+                        lease->leased = 1U;
+                        memcpy(&lease->addr, addr, sizeof(ipv6_addr_t));
+                        DEBUG("DHCPv6 client: ADD IP ADDRESS %s\n",
+                            ipv6_addr_to_str(addr_str, addr, sizeof(addr_str)));
+
+                        if (netif_set_opt(netif, NETOPT_IPV6_ADDR, 64 << 8, addr, sizeof(*addr)) > 0) {
```

Comment:
This is the current logic I used for adding IP addresses obtained via IA_NA to the net interface. Should this rather be moved somewhere else as it is the case with IA_PD and `dhcpv6_client_conf_prefix`?

Reply:
Yes. See https://github.com/RIOT-OS/RIOT/pull/16228#issuecomment-873427693.